src/sessions.ts

```diff
@@ -15,18 +15,30 @@ type BaseSessionResponse = {
   revoked_at: string,
   user_id: string,
   is_valid: boolean,
+  extendable_until: string,
+  max_extension_secs: number,
 }
 
 type LoginResponse = BaseResp & BaseSessionResponse
 type LogoutResponse = BaseResp & BaseSessionResponse
+type ExtendResponse = BaseResp & BaseSessionResponse
 type IsValidResponse = boolean
 type IsValidWithRoleResponse = boolean
 type SessionResponse = BaseResp & BaseSessionResponse
 
-function login(c: MalanConfig, username: string, password: string, expirationSeconds = 0): Promise<LoginResponse> {
+function login(c: MalanConfig, username: string, password: string, expirationSeconds = 0, maxExtensionSeconds = 0, extendableUntilSeconds = 0): Promise<LoginResponse> {
```

Comment:
positional arguments with identical types (number) aren't awesome in javascript but it looks like refactoring to an options object would already be a breaking change. this gets the job done without a major semver release. 👍